File: app/src/main/java/com/eslamdev/islamic/adapter/HadithAdapter.java
package com.eslamdev.islamic.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.eslamdev.islamic.R;
import com.eslamdev.islamic.model.Hadith;
import java.util.ArrayList;

public class HadithAdapter extends RecyclerView.Adapter<HadithAdapter.ViewHolder> {

    private final ArrayList<Hadith> hadithList;
    private final OnHadithClickListener listener;

    public interface OnHadithClickListener {
        void onHadithClick(Hadith hadith);
    }

    public HadithAdapter(ArrayList<Hadith> hadithList, OnHadithClickListener listener) {
        this.hadithList = hadithList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.hadith_item_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Hadith hadith = hadithList.get(position);
        holder.hadithId.setText(String.valueOf(position + 1));
        holder.hadithTitle.setText(hadith.getTitle());
        holder.itemView.setOnClickListener(v -> listener.onHadithClick(hadith));
    }

    @Override
    public int getItemCount() {
        return hadithList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView hadithId, hadithTitle;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            hadithId = itemView.findViewById(R.id.hadith_id);
            hadithTitle = itemView.findViewById(R.id.hadith_title);
        }
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/adapter/HadithAdapter.kt
package com.eslamdev.islamic.presentation.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.eslamdev.islamic.data.model.HadithModel
import com.eslamdev.islamic.databinding.HadithItemLayoutBinding // تأكد إن الـ XML اسمه كدا أو عدله

class HadithAdapter(private val onClick: (HadithModel) -> Unit) :
    ListAdapter<HadithModel, HadithAdapter.HadithViewHolder>(DiffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HadithViewHolder {
        val binding = HadithItemLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return HadithViewHolder(binding)
    }

    override fun onBindViewHolder(holder: HadithViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }

    inner class HadithViewHolder(private val binding: HadithItemLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(hadith: HadithModel) {
            binding.hadithId.text = "${position + 1}"
            binding.hadithTitle.text = hadith.title
            binding.root.setOnClickListener { onClick(hadith) }
        }
    }

    companion object DiffCallback : DiffUtil.ItemCallback<HadithModel>() {
        override fun areItemsTheSame(oldItem: HadithModel, newItem: HadithModel) = oldItem.id == newItem.id
        override fun areContentsTheSame(oldItem: HadithModel, newItem: HadithModel) = oldItem == newItem
    }
}